examples/plot_connectivity.py

```diff
@@ -30,73 +33,99 @@
+
+# Pyramidal cell connections
+location, receptor = 'distal', 'ampa'
+weight, delay, lamtha = 1.0, 1.0, 70
+src = 'L5_pyramidal'
+for target in ['L5_pyramidal', 'L5_pyramidal']:
+    net_all.add_connection(src, target, location, receptor,
+                           delay, weight, lamtha)
+
+# Basket cell connections
+location, receptor = 'soma', 'gabaa'
+weight, delay, lamtha = 1.0, 1.0, 70
+src = 'L2_basket'
+for target in ['L2_basket', 'L5_pyramidal']:
+    net_all.add_connection(src, target, location, receptor,
+                           delay, weight, lamtha)
+
+dpl_all = simulate_dipole(net_all, n_trials=1)
+net_all.cell_response.plot_spikes_raster()
+
+###############################################################################
+# That's a lot of spiking! We can additionally use the ``probability``.
+# argument to create a sparse connectivity pattern instead of all-to-all. Let's
+# try creating the same network with a 10% change of cells connecting
+# to each other. The resulting connectivity pattern can also be visualized
+# with ``net.connectivity[idx].plot()``
+probability = 0.1
+net_sparse = Network(params, add_drives_from_params=True)
+
+# Pyramidal cell connections
+location, receptor = 'distal', 'ampa'
+weight, delay, lamtha = 1.0, 1.0, 70
+src = 'L5_pyramidal'
+for target in ['L5_pyramidal', 'L2_basket']:
+    net_sparse.add_connection(src, target, location, receptor,
+                              delay, weight, lamtha, probability)
+
+# Basket cell connections
 location, receptor = 'soma', 'gabaa'
 weight, delay, lamtha = 1.0, 1.0, 70
-net.add_connection(src_gid, target_gids, location, receptor,
-                   delay, weight, lamtha)
+src = 'L2_basket'
+for target in ['L2_basket', 'L5_pyramidal']:
+    net_sparse.add_connection(src, target, location, receptor,
+                              delay, weight, lamtha, probability)
 
-net_add = net.copy()
-dpl_add = simulate_dipole(net_add, n_trials=1)
-net_add.cell_response.plot_spikes_raster()
+dpl_sparse = simulate_dipole(net_sparse, n_trials=1)
+net_sparse.cell_response.plot_spikes_raster()
+net_sparse.connectivity[-2].plot()
+net_sparse.connectivity[-1].plot()
 
 ###############################################################################
-# Adding more inhibitory connections did not completely restore the normal
-# spiking. L2 basket and pyramidal cells rhythymically fire in the gamma
-# range (30-80 Hz). As a final step, we can see how this change in spiking
-# activity impacts the aggregate current dipole.
+# Using the sparse connectivity pattern led produced a lot more spiking in
+# the L5 pyramidal cells. While there are less excitatory connections overall,
+# there was also a decrease in the inhibtory connections. This shift in
+# activity is often referred to as the excitatory/inhibitory (E/I) balance.
+# As a final step, we can see how this change in spiking activity impacts
+# the aggregate current dipole.
```

Comment:
okay I am confused which way the E/I balance shifted?

Also, it might be more intuitive for the reader to find this hypothesis first before you try changing the connectivity, then what you tried, and finally your observation confirming the hypothesis

Reply:
I was trying to draw attention to how E/I balance is not just a product of the raw number of connections, but also how the circuit is wired. Honestly this is a bit too much for a simple example so I am just adding descriptions of the resulting spiking activity.